begin/cmdline.py

```diff
@@ -56,18 +57,44 @@ def from_param(self, param, default=NODEFAULT):
 
     def from_name(self, name, default=NODEFAULT, section=None):
         "Get default value from argument name"
-        if len(self._parser.sections()) > 0:
-            section = self._section if section is None else section
-            try:
-                default = self._parser.get(section, name)
-            except (configparser.NoSectionError, configparser.NoOptionError):
-                pass
+        sections = self._get_list_section(section) if section is not None else self.section
+        if len(sections) > 0:
```

Comment:
Why is this needed? Won't the for loop just do nothing if there are no sections?